addons/stock_picking_batch/models/stock_picking.py

```diff
@@ -86,3 +125,77 @@ def _should_show_transfers(self):
         if len(self.batch_id) == 1 and self == self.batch_id.picking_ids:
             return False
         return super()._should_show_transfers()
+
+    def _find_auto_batch(self):
+        # Search for other pickings that shares the same auto_batch criteria
+        domain = self._get_auto_batch_domain()
+        # By ordering pickings by their batch_id, we try to put the picking into an existing batch before trying to make a new one.
+        batchable_pickings = self.env['stock.picking'].search(domain, order="batch_id ASC")
+        full_batch_ids = set()
+        for picking in batchable_pickings:
+            if picking.batch_id and picking.batch_id.id not in full_batch_ids:
+                if picking.batch_id._is_picking_auto_mergeable(self):
+                    # Add picking to batch
+                    picking.batch_id.picking_ids |= self
+                    return picking.batch_id
+                else:
+                    full_batch_ids.add(picking.batch_id.id)
+            if not picking.batch_id and self._is_auto_batchable(picking):
+                # Create new batch with both pickings
+                new_batch = self.env['stock.picking.batch'].create({
+                    'picking_ids': [(4, self.id), (4, picking.id)],
+                    'company_id': self.company_id.id if self.company_id else False,
+                    'picking_type_id': self.picking_type_id.id,
+                })
+                if picking.picking_type_id.batch_auto_confirm:
+                    new_batch.action_confirm()
```

Comment:
I think this creates a confusing behavior. It makes it so when creating a new picking that can be auto-batched, when you save the picking then it immediately becomes confirmed. Maybe that's ok, but I would think it's annoying when a user wants to make a draft transfer and have it auto-batch only once its confirmed...

Reply:
You're right on that part, but auto-batch trigger will be moved to the picking's action_confirm instead of create, so I don't think it will be a problem anymore.